### PR TITLE
Add wallet configuration for bitcoin core 24

### PR DESCRIPTION
**Pull request by rswol**
### Description

Enable custom BTC wallet for operating in mining mode 

### Applicable issues
- fixes #

### Additional info (benefits, drawbacks, caveats)

Bitcoin core 24 doesn't support default wallet any more. This change is necessary to enable future support of current bitcoin core nodes.

### Checklist
- [x] Test coverage for new or modified code paths
- [N/A] Changelog is updated
- [N/A] Required documentation changes (e.g., `docs/rpc/openapi.yaml` and `rpc-endpoints.md` for v2 endpoints, `event-dispatcher.md` for new events)
- [N/A] New clarity functions have corresponding PR in `clarity-benchmarking` repo
- [N/A] New integration test(s) added to `bitcoin-tests.yml`

### CODE REVIEW

**Comment by pavitthrap** [comment]
Hi, thanks for contributing to stacks-blockchain!

File: testnet/stacks-node/src/config.rs [hunk 1]
```diff
@@ -872,6 +872,7 @@ impl Config {
                 unprocessed_block_deadline_secs: miner
                     .unprocessed_block_deadline_secs
                     .unwrap_or(miner_default_config.unprocessed_block_deadline_secs),
+                wallet: miner.wallet.as_ref().map(|c| { c.clone() }).unwrap_or("".to_string()),
```
Comment (pavitthrap) [inline testnet/stacks-node/src/config.rs:875]: I would change this to `miner.wallet.clone().unwrap_or("".to_string());`

File: testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs [hunk 2]
```diff
@@ -2138,7 +2138,7 @@ impl BitcoinRPCRequest {
         let url = {
             // some methods require a wallet ID
             let wallet_id = match payload.method.as_str() {
-                "importaddress" | "listunspent" => Some("".to_string()),
+                "importaddress" | "listunspent" => Some(config.miner.wallet.clone()),
```
Comment (pavitthrap) [inline testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs:2141]: I would consider renaming "wallet" to "wallet_id" throughout the PR.

**Review by pavitthrap — APPROVED** [review]
I had two small non-blocking comments.

**Comment by wileyj** [comment]
@rswol 
The proposed changes look fine to me, but I'd like to see some of the docs updated to handle this new config setting. 
example: 
https://github.com/stacks-network/stacks-blockchain/blob/master/docs/mining.md
https://github.com/stacks-network/stacks-blockchain/blob/master/testnet/stacks-node/conf/mainnet-miner-conf.toml
https://github.com/stacks-network/stacks-blockchain/blob/master/testnet/stacks-node/conf/testnet-miner-conf.toml

**Comment by pavitthrap** [comment]
I'm about to merge changes from master into develop. You'll probably need to rebase your PR after that merge is completed. 

I can drop a note in here once that merge is complete.

**Comment by pavitthrap** [comment]
The merge is complete, you can go ahead and rebase on top of develop!

**Pull request closed without merging**